src/plugins/intel_gpu/src/graph/impls/onednn/fully_connected_onednn.cpp

```diff
@@ -130,25 +130,71 @@ struct fully_connected_onednn : typed_primitive_onednn_impl<fully_connected> {
         get_matmul_primitive_descriptor(const kernel_impl_params& impl_params,
                                         cldnn::engine& engine,
                                         size_t prim_input_size,
+                                        size_t prim_weights_rank,
                                         bool has_bias,
                                         const dnnl::primitive_attr& attr = dnnl::primitive_attr()) {
         auto input_layout = impl_params.get_input_layout(0);
         auto weights_layout = impl_params.get_input_layout(1);
         auto output_layout = impl_params.get_output_layout();
 
-        transform_layouts(input_layout, weights_layout, output_layout, prim_input_size);
+        dnnl::memory::format_tag target_fmt;
+        dnnl::memory::format_tag weights_fmt;
+
+        if (prim_input_size == 3) {
+            target_fmt = dnnl::memory::format_tag::abc;
+            weights_fmt = dnnl::memory::format_tag::acb;
+        } else if (prim_input_size == 4) {
+            target_fmt = dnnl::memory::format_tag::abcd;
+            weights_fmt = dnnl::memory::format_tag::abdc;
+        } else if (prim_input_size == 5) {
+            target_fmt = dnnl::memory::format_tag::abcde;
+            weights_fmt = dnnl::memory::format_tag::abced;
+        } else if (prim_input_size == 6) {
+            target_fmt = dnnl::memory::format_tag::abcdef;
+            weights_fmt = dnnl::memory::format_tag::abcdfe;
+        } else {
+            target_fmt = dnnl::memory::format_tag::ab;
+            weights_fmt = dnnl::memory::format_tag::ba;
+        }
+
+        // Transform weights_layout according to input layout
+        {
+            ov::PartialShape new_weights_pshape;
 
-        auto input_md = onednn::layout_to_memory_desc(input_layout, dnnl::memory::format_tag::ab, false);
-        // TODO: should change format to any. May need a reorder.
-        auto weights_md = onednn::layout_to_memory_desc(weights_layout, dnnl::memory::format_tag::ba);
-        auto output_md = onednn::layout_to_memory_desc(output_layout, dnnl::memory::format_tag::ab, false);
+            for (size_t i = 0; i < (prim_input_size - prim_weights_rank); i++) {
+                new_weights_pshape.push_back(1);
+            }
+
+            for (size_t i = 0; i < prim_weights_rank; i++) {
+                new_weights_pshape.push_back(weights_layout.get_partial_shape()[i]);
+            }
```

Comment:
This code assumes that the broadcast will happen only on the outermost axis, right?

Reply:
Yes, right.